### PR TITLE
[RDF] Fix build warnings introduced in TakeHelper

### PR DESCRIPTION
**Pull request by stwunsch**
See here: https://cdash.cern.ch/viewBuildError.php?type=1&buildid=604376

### CODE REVIEW

**Comment by stwunsch** [comment]
Somehow it's not good for jenkins ;) The overload of the free function is not picked up with the universal ref.

**Comment by stwunsch** [comment]
@dpiparo Do I need to make the specialization like this? ->

```c++
template <typename COLL>
void FillColl(bool&& v, COLL& c) {
   c.push_back(v);
}
```

Shouldn't `void FillColl(bool v, COLL& c)` also be picked up as "more specialized"?

**Comment by stwunsch** [comment]
The behaviour seems to be compiler dependent. On my machine (gcc7), it is picked up.

**Comment by stwunsch** [comment]
I love C++ :D Full of interesting features:

```cpp
#include <vector>
#include <iostream>

template <typename T, typename U>
void Fill(T&& v, U& u) {
    std::cout << "template! " << v << std::endl;
}
template <typename U>
void Fill(bool v, U& u) {
    std::cout << "bool! " << v << std::endl;
}

template <typename T>
void Loop(std::vector<T>& coll) {
    for(auto&& v : coll) {      // <-- Replace auto by T, then it takes the "bool" version if Fill and not the "template" one (see above)
        Fill(v, coll);
    }
}

int main(){
    std::vector<bool> coll = {0, 1, 0};
    Loop(coll);
}
```

**Comment by stwunsch** [comment]
@phsft-bot build

**Comment by stwunsch** [comment]
@phsft-bot build

**Comment by stwunsch** [comment]
@dpiparo I've put here an implementation of the fix you suggested for the `RVec<bool>` issue.

File: math/vecops/inc/ROOT/RVec.hxx [hunk 1]
```diff
@@ -40,6 +40,21 @@
 namespace ROOT {
 
 namespace VecOps {
+
+namespace Internal {
+
+template <typename... Args>
+void REmplaceBackHelper(std::vector<bool>& v, Args... args) {
+   v.push_back(std::forward<Args>(args)...);
```
Comment (bluehood) [inline math/vecops/inc/ROOT/RVec.hxx]: here and (especially, because of move semantics) below, needs a `std::forward<Args>(args)...`

Comment (stwunsch) [reply math/vecops/inc/ROOT/RVec.hxx]: but there is one? what do i miss?

Comment (bluehood) [reply math/vecops/inc/ROOT/RVec.hxx]: pfff nothing sorry, am tired...

**Comment by stwunsch** [comment]
@phsft-bot build

**Comment by stwunsch** [comment]
@phsft-bot build

**Pull request merged**